src/ipc/ipc4/dai.c

```diff
@@ -31,6 +31,44 @@
 
 LOG_MODULE_DECLARE(ipc, CONFIG_SOF_LOG_LEVEL);
 
+int dai_set_link_hda_config(uint16_t *link_config,
+			struct ipc_config_dai *common_config,
+			const void *spec_config)
+{
+#if defined(CONFIG_ACE_VERSION_2_0)
+	struct ipc4_audio_format *out_fmt = (struct ipc4_audio_format *) common_config->out_fmt;
+	struct sof_ipc_dai_config *sof_cfg;
+	union hdalink_cfg link_cfg;
+
+	switch (common_config->type) {
+	case SOF_DAI_INTEL_SSP:
+		sof_cfg = (struct sof_ipc_dai_config *)spec_config;
```

Comment:
this also drops the `const` qualifier - another reason to avoid type-casts wherever possible